include/modem/lte_lc.h

```diff
@@ -214,21 +214,25 @@ enum lte_lc_evt_type {
 	 */
 	LTE_LC_EVT_NW_REG_STATUS,
 
+#if defined(CONFIG_LTE_LC_PSM)
 	/**
 	 * PSM parameters provided by the network.
 	 *
 	 * The associated payload is the @c lte_lc_evt.psm_cfg member of type
 	 * @ref lte_lc_psm_cfg in the event.
 	 */
 	LTE_LC_EVT_PSM_UPDATE,
```

Comment:
Good point

Reply:
Thanks, some nitpicking, though. The numbering starts from 1, while earlier it has started from 0. 😃

Reply:
I took as example other similar enums. The way I see it, starting from 0 makes sense when each value identifies an "option" while starting from 1 makes more sense when each value identifies a variant of an enum type.

Reply:
Yes, I don't mind that the numbering starts from 1 instead of 0 as such, but just wanted to highlight that this PR changes the numbering. I guess this is not a problem, at least the numbering is now static regardless of the configuration.

Reply:
Ah, I see what you mean now, I don't have any problem changing it to starting from 0 then.

Reply:
@tokangas 
Why do you rely on value of the enum?
If that is needed, then why don't you just remove the #ifdeffing...

You guys seem to spend quite a bit of effort with intention to break builds with wrong configuration.

Reply:
@SeppoTakalo For example when doing unit tests, you see those event IDs in test output and after a while recognize the event by the number. If there's no reason to change the values, I would like to keep them as they were. I fail to understand how keeping the numbering intact breaks any builds.

Reply:
No it is the `#ifdef THIS` that breaks the build, like I commented earlier in this same PR.

Reply:
@SeppoTakalo This update gives back to the user more available flash space when they are using just the core features. As of now (in main) each application using this library is building multiple components that are most of the time never used (all of the AT monitors are always compiled even when not used, which is a waste of flash). Having to enable some Kconfigs to enable the extra functionalities for some existing applications and for our CI is just an unavoidable consequence.